x-pack/legacy/plugins/lens/public/xy_visualization_plugin/xy_config_panel.tsx

```diff
@@ -94,67 +84,161 @@ function newLayerState(layerId: string): LayerConfig {
-              label={i18n.translate('xpack.lens.xyChart.layerLabel', {
-                defaultMessage: 'Layer',
-              })}
+            <EuiPopover
+              id="lnsXY_layer"
+              isOpen={localState.openLayerId === layer.layerId}
+              closePopover={() => {
+                setLocalState({ ...localState, openLayerId: null });
+              }}
+              className="eui-fullWidth"
+              anchorClassName="eui-fullWidth"
+              button={
+                <EuiFormRow>
+                  <EuiFlexGroup>
+                    <EuiFlexItem>
+                      <EuiButtonIcon
+                        iconType="gear"
+                        data-test-subj="lnsXY_layer_advanced"
+                        aria-label={i18n.translate('xpack.lens.xyChart.layerSettings', {
+                          defaultMessage: 'Edit layer settings',
+                        })}
+                        onClick={() => {
+                          setLocalState({ ...localState, openLayerId: layer.layerId });
+                        }}
+                      />
+                    </EuiFlexItem>
+
+                    <EuiFlexItem>
+                      <EuiIcon
+                        type={chartTypeIcons.find(icon => icon.id === layer.seriesType)!.iconType}
+                      />
+                    </EuiFlexItem>
+
+                    <EuiFlexItem>
+                      <NativeRenderer
+                        data-test-subj="lnsXY_layerHeader"
+                        render={props.frame.datasourceLayers[layer.layerId].renderLayerPanel}
+                        nativeProps={{ layerId: layer.layerId }}
+                      />
+                    </EuiFlexItem>
+                  </EuiFlexGroup>
+                </EuiFormRow>
+              }
             >
               <>
-                <EuiButtonIcon
+                <EuiFormRow
+                  label={i18n.translate('xpack.lens.xyChart.chartTypeLabel', {
+                    defaultMessage: 'Chart type',
+                  })}
+                >
+                  <EuiButtonGroup
+                    legend={i18n.translate('xpack.lens.xyChart.chartTypeLegend', {
+                      defaultMessage: 'Chart type',
+                    })}
+                    name="chartType"
+                    className="eui-displayInlineBlock"
+                    data-test-subj="lnsXY_seriesType"
+                    options={chartTypeIcons}
+                    idSelected={layer.seriesType}
+                    onChange={seriesType => {
+                      setState(
+                        updateLayer(
+                          state,
+                          { ...layer, seriesType: seriesType as SeriesType },
+                          index
+                        )
+                      );
+                    }}
+                    isIconOnly
+                  />
+                </EuiFormRow>
+
+                <EuiButton
```

Comment:
Can we be sure this button only appears if there are multiple layers? Makes sense that you shouldn't be able to delete a layer if it's the only one.

Reply:
That's probably a good idea. In other places (e.g. the y configuration), we let you delete it, but we automatically add a new one, so it essentially serves as a "clear" in that scenario.

Reply:
I'll explore this change in a separate PR, the main issue here is that layers are currently attached to the index pattern that was used when creating them, so I have found it useful to have a totally unconfigured state so I can reset everything.